docs/dev/pass_infra.rst

```diff
@@ -123,16 +124,23 @@ Python APIs to create a compilation pipeline using pass context.
 
     class PassContextNode : public Object {
      public:
-      ErrorReporter err_reporter;
```

Comment:
It seems to be replaced by diag_ctx; in #6274.
It is not related to this PR. I fixed it just because it's handy.
Though there aren't lots of descriptions about error reporters in pass_infra.txt, I think it should present due to this sentence:
> PassContext carries useful information for an optimization pass. For example, it contains the error reporting system so optimization authors can provide diagnostic...